go.mod

```diff
@@ -85,29 +86,29 @@ require (
 	// open telemetry dependencies
 	// pinned to v0.97.0 due to issue with ports binding which causes our tests to fail
 	// Do NOT upgrade the version beyond 0.97.0 until https://github.com/open-telemetry/opentelemetry-collector/issues/10031 is fixed!
```

Comment:
https://github.com/open-telemetry/opentelemetry-collector/issues/10031 has been resolved via https://github.com/open-telemetry/opentelemetry-collector/pull/10078 and it looks like that fix was included in the `v0.100.0` release of the OTel Collector.  So it should be safe to remove these comments and proceed with the version update in this PR.